torch/_subclasses/fake_tensor.py

```diff
@@ -369,7 +374,18 @@ def mk_fake_tensor(make_meta_t):
             source=source,
             dynamic_dims=dynamic_dims,
             constraint_dims=constraint_dims,
+            allocate_fresh_if_metadata_mutated=allocate_fresh_if_metadata_mutated,
         )
+        # Bind original shape and stride to the FakeTensor
+        # We must do this because as we trace, transforms like setting .data will
+        # change the fake tensor. This means that if we do not keep track of the original meta,
+        # we will lose it.
+        # This matters, because there will be points where we enter tracing
+        # (like aot_autograd after dynamo), where accessing memo will give us the tensor
+        # as it is at the end of trace, but we need to get the tensor as it was before tracing.
+        out._original_sym_shape = out.size()
```

Comment:
Would it makes sense for this behavior to always be true?

Reply:
Good question, let me think about it :)